### PR TITLE
Expose close() method from PipelineClient in Storage Blobs

### PR DESCRIPTION
**Pull request by rakshith91**
Fixes Azure#8564

- Regenerate _generated code.
- Expose close() in Mixin

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_generated/aio/operations_async/_block_blob_operations_async.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         self._config = config
         self.x_ms_blob_type = "BlockBlob"
 
-    async def upload(self, body, content_length, timeout=None, metadata=None, tier=None, request_id=None, blob_http_headers=None, lease_access_conditions=None, cpk_info=None, modified_access_conditions=None, *, cls=None, **kwargs):
+    async def upload(self, body, content_length, timeout=None, transactional_content_md5=None, metadata=None, tier=None, request_id=None, blob_http_headers=None, lease_access_conditions=None, cpk_info=None, modified_access_conditions=None, *, cls=None, **kwargs):
```
Comment (kristapratico) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_generated/aio/operations_async/_block_blob_operations_async.py]: are we exposing this as a kwarg to the user?

Comment (rakshith91) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_generated/aio/operations_async/_block_blob_operations_async.py]: no

File: sdk/storage/azure-storage-blob/tests/test_blob_client.py [hunk 2]
```diff
@@ -574,5 +574,14 @@ def test_error_with_malformed_conn_str(self):
...
+    @GlobalStorageAccountPreparer()
+    def test_closing_pipeline_client(self, resource_group, location, storage_account, storage_account_key):
+        # Arrange
+        for client, url in SERVICES.items():
+            # Act
+            service = client(
+                self.account_url(storage_account.name, "blob"), credential=storage_account_key, container_name='foo', blob_name='bar')
 
+            # Assert
+            assert hasattr(service, 'close')
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_blob_client.py]: you could actually call it too :)

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/operations/_share_operations.py [hunk 3]
```diff
@@ -395,8 +395,7 @@ def get_permission(self, file_permission_key=None, timeout=None, cls=None, **kwa
         # Construct headers
         header_parameters = {}
         header_parameters['Accept'] = 'application/json'
-        if file_permission_key is not None:
-            header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
+        header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
```
Comment (lmazuel) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/operations/_share_operations.py]: This means permissions is now required, did you check this doesn't break anything and our surface API already was making this required?

File: sdk/storage/azure-storage-file-share/tests/test_file_client.py [hunk 4]
```diff
@@ -417,6 +417,16 @@ def test_error_with_malformed_conn_str(self):
...
+    def test_closing_pipeline_client_async(self):
+        # Arrange
+
+        for client, url in SERVICES.items():
+            # Act
+            service = client(
+                self.get_file_url(), credential=self.account_key, share_name='foo', directory_path='bar', file_path='baz')
+
+            # Assert
+            assert hasattr(service, 'close')
```
Comment (lmazuel) [inline sdk/storage/azure-storage-file-share/tests/test_file_client.py]: call it! :)

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/aio/operations_async/_share_operations_async.py [hunk 5]
```diff
@@ -395,8 +395,7 @@ def __init__(self, client, config, serializer, deserializer) -> None:
         # Construct headers
         header_parameters = {}
         header_parameters['Accept'] = 'application/json'
-        if file_permission_key is not None:
-            header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
+        header_parameters['x-ms-file-permission-key'] = self._serialize.header("file_permission_key", file_permission_key, 'str')
```
Comment (lmazuel) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/aio/operations_async/_share_operations_async.py]: Breaking change: please check our surface API was already making this guy required, or at least was always fitting a default if not provided by customer

File: sdk/storage/azure-storage-queue/tests/test_queue_client_async.py [hunk 6]
```diff
@@ -470,6 +470,17 @@ def callback(response):
...
+    @GlobalStorageAccountPreparer()
+    def test_closing_pipeline_client(self, resource_group, location, storage_account, storage_account_key):
+        # Arrange
+        for client, url in SERVICES.items():
+            # Act
+            service = client(
+                self.account_url(storage_account.name, "queue"), credential=storage_account_key, queue_name='queue')
+
+            # Assert
+            exists = hasattr(service, 'close')
```
Comment (lmazuel) [inline sdk/storage/azure-storage-queue/tests/test_queue_client_async.py]: why the "exists" variable?

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/operations/_share_operations.py (hunk 3 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/operations/_share_operations.py]: yes, it's required.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/aio/operations_async/_share_operations_async.py (hunk 5 above)
Comment (rakshith91) [reply sdk/storage/azure-storage-file-share/azure/storage/fileshare/_generated/aio/operations_async/_share_operations_async.py]: yes, it's required

File: sdk/storage/azure-storage-blob/tests/test_blob_client.py [hunk 7]
```diff
@@ -33,7 +33,7 @@
 class StorageClientTest(StorageTestCase):
     def setUp(self):
         super(StorageClientTest, self).setUp()
-        self.sas_token = '?sv=2015-04-05&st=2015-04-29T22%3A18%3A26Z&se=2015-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=Z%2FRHIX5Xcg0Mq2rqI3OlWTjEg2tYkboXr1P9ZUXDtkk%3D'
+        self.sas_token = '?sv=2015-04-05&st=2015-04-29T22%3A18%3A26Z&se=2015-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=Y%2FoUshaLLnoTPass'
```
Comment (jongio) [inline sdk/storage/azure-storage-blob/tests/test_blob_client.py]: Do we need to have any of the Sas_Token in code?  I'm not a huge fan of the "shallnotpass" text...kind of mocking tone.   Can we just do:

`self.sas_token = ''`

And then inject the new one at runtime?

This comment applies to all sas_tokens in this PR.

File: sdk/storage/azure-storage-blob/tests/test_blob_client_async.py [hunk 8]
```diff
@@ -548,4 +548,17 @@ def callback(response):
...
+
+        for client, url in SERVICES.items():
+            # Act
+            service = client(
+                self.account_url(storage_account.name, "blob"), credential=storage_account_key, container_name='foo', blob_name='bar')
+
+            # Assert
+            async with service:
+                assert hasattr(service, 'close')
+                service.close()
```
Comment (lmazuel) [inline sdk/storage/azure-storage-blob/tests/test_blob_client_async.py]: Side note, the "with" call close too, so this test is technically more stresfull that it should be, it closes the service, and then exit the contextmanager and tries to re-close it.
It's nice to see that the second close doesn't break it, but it's not really the nominal test :).

For sake of clarity I would do a simple "create and and close" and keep your stress test "create, enter context manager, close, close context manager"

**Pull request merged**